packages/react/src/createReactQueryHooks.tsx

```diff
@@ -320,12 +320,17 @@ export function createReactQueryHooks<
     ) {
       void prefetchQuery(pathAndInput as any, opts as any);
     }
-    const actualOpts = useSSRQueryOptionsIfNeeded(pathAndInput, opts);
+    const ssrOpts = useSSRQueryOptionsIfNeeded(pathAndInput, opts);
 
     return __useQuery(
       pathAndInput as any,
-      () => (client as any).query(...getClientArgs(pathAndInput, actualOpts)),
-      actualOpts,
+      ({ signal }) => {
+        const actualOpts = { ...ssrOpts, trpc: { ...ssrOpts?.trpc, signal } };
+        return (client as any).query(
+          ...getClientArgs(pathAndInput, actualOpts),
```

Comment:
I think we made a mistake when changing trpc properties to be under the `trpc` object (this also applies to `useMutation`)
```suggestion
          ...getClientArgs(pathAndInput, actualOpts.trpc),
```

Reply:
why would we only want to pass in the trpc options here?

Reply:
Those are the options meant to be passed to `client.query`. Otherwise, we'll just be passing React Query options to the vanilla client.

Reply:
Just confirmed, the `getClientArgs` flattens the options:

https://github.com/trpc/trpc/blob/98a774098a1b46ffd126ded905d53c26204a808e/packages/react/src/createReactQueryHooks.tsx#L95-L101